creator-node/src/components/tracks/TrackTranscodeHandoffManager.js

```diff
@@ -557,6 +549,50 @@ class TrackTranscodeHandoffManager {
 
     return TrackTranscodeHandoffManager.logger
   }
+
+  /**
+   * Wrapper around async-retry API, with no retry on 404. Used to handle backwards compatibility when
+   * an SP has not yet upgraded and the route is not found.
+   *
+   * options described here https://github.com/tim-kos/node-retry#retrytimeoutsoptions
+   * @param {Object} param
+   * @param {Object} param.logger
+   * @param {func} param.asyncFn the fn to asynchronously retry
+   * @param {string} param.asyncFnLabel the task label used to print on retry. used for debugging purposes
+   * @param {Object} param.options optional options. defaults to the params listed below if not explicitly passed in
+   * @param {number} [param.options.factor=2] the exponential factor
+   * @param {number} [param.options.retries=5] the max number of retries. defaulted to 5
+   * @param {number} [param.options.minTimeout=1000] minimum number of ms to wait after first retry. defaulted to 1000ms
+   * @param {number} [param.options.maxTimeout=5000] maximum number of ms between two retries. defaulted to 5000ms
+   * @param {func} [param.options.onRetry] fn that gets called per retry
+   * @returns the fn response if success, or throws an error
+   */
+  static asyncRetryNotOn404({
+    logger,
+    asyncFn: inputAsyncFn,
+    asyncFnLabel,
+    options = {}
+  }) {
+    const asyncFn = async (bail) => {
+      let resp
+      try {
+        resp = await inputAsyncFn()
```

Comment:
nit i think you could just do `return inputAsyncFn()` here instead of `let resp` and `return resp` below

Reply:
I dont think we can do that bc we need to catch and see if the status code is a 404 to bail out early

Reply:
ah see this abstraction is actually confusing bc the status code being 404 will only happen if the asyncFn is a direct axios call
my comment above was also assuming that it was a direct axios call, in which case it will throw an error on any non 200 status resp

if we want to keep the 404 wrapper, it should be specific to axios calls and make that call internally - can send you code example in a bit

also - in interest of shipping this, we can address after merge